src/components/Form/RecipeForm.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useRef } from 'react';
+import { useDispatch } from 'react-redux';
+
+import Ingredients from '../Ingredients/Ingredients';
+import { editRecipe } from '../../store/recipes/actions';
+import { IRecipe } from '../../store/interfaces';
+
+const RecipeForm = ({ id, title, preparation, tips, ingredients }: IRecipe) => {
+  const isEditMode = true;
+  const [titleInput, setTitleInput] = useState(title);
+  const [preparationInput, setPreparationInput] = useState(preparation);
+  const [tipsInput, setTipsInput] = useState(tips);
+
+  const dispatch = useDispatch();
+  const componentMounted = useRef(false);
+
+  const handleTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => setTitleInput(e.target.value);
+  const handlePreparationChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => setPreparationInput(e.target.value);
+  const handleTipsChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => setTipsInput(e.target.value);
+
+  const updateRecipe = (id: number | string, changedRecipe: IRecipe) => dispatch(editRecipe({ id, changedRecipe }));
+
+  let ingredientsToUpdate = ingredients;
+
+  const handleFormSubmit = (id: number | string) => async () => {
+    // e.preventDefault();
+
+    componentMounted.current = true;
```

Comment:
Sprawdzanie stanu zamontowania komponentu może być przydatne w szczególnych przypadkach, np. w funkcjach wywoływanych na setTimeoucie. Tutaj nie ma takiego uzasadnienia, więc jest to zbędne.